controller_manager:
  ros__parameters:
    update_rate: 100

    leader_mpc:
      type: formation_mpc/LeaderMPC

    leader_joint_bcast:
      type: joint_state_broadcaster/JointStateBroadcaster

leader_joint_bcast:
  ros__parameters:
    use_local_topics: True
    joints:
      - "joint_x"
      - "joint_y"
      - "joint_rz"
      - "ur/shoulder_pan_joint"
      - "ur/shoulder_lift_joint"
      - "ur/elbow_joint"
      - "ur/wrist_1_joint"
      - "ur/wrist_2_joint"
      - "ur/wrist_3_joint"

leader_mpc: # TODO: guarda ??
  ros__parameters:
    control_horizon_in_ms: 50.0
    prediction_horizon: 5
    number_of_points: 2
    dt: 0.01
    formation_dt: 0.1
    click_gain: 1
    pid:
      P: [1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0]
    action_monitor_rate: 100.0
    leader_ns: "sweepee_1"
    is_omni: false
    mobile_base_link: "base_link"
    manipulator_base_link: "ur/base_link"
    tool_frame: "ur/tool0"
    map_frame: "map"
    mount_frame: "mount_frame"
    joints:
      - "joint_x"
      - "joint_y"
      - "joint_rz"
      - "ur/shoulder_pan_joint"
      - "ur/shoulder_lift_joint"
      - "ur/elbow_joint"
      - "ur/wrist_1_joint"
      - "ur/wrist_2_joint"
      - "ur/wrist_3_joint"
    initial_positions:
      - 0.0
      - 0.0
      - 0.0
      - 0.0
      - -2.356
      - 1.57
      - 0.0
      - 0.0
      - 0.0
    command_interface:
#      - "velocity"
      - "position"
    state_interface:
      - "position"
      - "velocity"
    # robot_description_topic: "robot_description"
    # robot_description: ## from launcher
    cmd_vel_topic: "cmd_vel"
    # cmd_trj_topic: ## default
    scaling:
      active: true
      limit: 2.0
    hierarchy: [1,2,3]
    trj_leader_topic: "trj_share"
    plan_topic: "/plan"
    base_max_vel: [2.0, 2.0, 2.0, 5.0, 5.0, 5.0]
    fake_base:
      package: "cooperative_transport_cell"
      path: "urdf/fake_base.urdf"
